rclpy/rclpy/node.py

```diff
@@ -562,6 +560,8 @@ def get_parameters(self, names: List[str]) -> List[Parameter]:
             undeclared parameters are allowed.
         :raises: ParameterNotDeclaredException if undeclared parameters are not allowed,
             and at least one parameter hadn't been declared beforehand.
+        :raises: ParameterUninitializedException if at least one parameter is statically typed and
```

Comment:
should that exception be raised?

I don't remember what rclcpp is exactly throwing, undeclaring a parameter was invalid when they are statically typed.
The exception should be the same no matter if it's uninitialzed or not, and shouldn't be `ParameterUninitializedException`.

It would be great to test this in both `rclcpp` and here

Reply:
I'm not sure I follow. I think this mimics the behavior of rclcpp. If a user calls `get_parameters(['foo', 'bar'])` and `bar` is declared, but not initialized, then `ParameterUninitializedException` is raised. Would you expect something else?

Reply:
lol, I thought this was the documentation of `undeclare_parameter()` sorry